libraries/core_libs/consensus/src/transaction/transaction_queue.cpp

```diff
@@ -52,18 +72,51 @@ bool TransactionQueue::erase(const trx_hash_t &hash) {
   return true;
 }
 
-bool TransactionQueue::insert(std::shared_ptr<Transaction> &&transaction) {
-  assert(transaction);
-  if (hash_queue_.contains(transaction->getHash())) return false;
+bool TransactionQueue::insert(std::pair<std::shared_ptr<Transaction>, TransactionStatus> &&transaction,
+                              uint64_t last_block_number) {
+  assert(transaction.first);
+  if (contains(transaction.first->getHash())) return false;
 
-  const auto tx_hash = transaction->getHash();
-  const auto it = priority_queue_.insert(std::move(transaction));
+  const auto tx_hash = transaction.first->getHash();
```

Comment:
if you do this line before it would be cleaner

Reply:
Fixed